### PR TITLE
fix: withdrawal double spend bug

### LINKED ISSUE #545: Withdrawal bug
- [x] Fix the [todo](https://github.com/Polkadex-Substrate/Polkadex/blob/f299d744deabb7830d97347d0ad8e48387661242/pallets/ocex/src/lib.rs#L742)

### PR DESCRIPTION
**Pull request by felixfaisal**
This PR resolves #545

### CODE REVIEW

File: pallets/ocex/src/lib.rs [hunk 1]
```diff
@@ -734,45 +735,72 @@ pub mod pallet {
...
+									.is_ok()
+									{
+										processed_withdrawals
+											.try_push(withdrawal)
+											.unwrap_or_default();
+									}
+								}
+							}
+						}
+						btree_map.remove(&account);
```
Comment (Gauthamastro) [inline pallets/ocex/src/lib.rs]: Keep the failed withdrawals for later inspection.

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

File: pallets/ocex/src/lib.rs (hunk 1 above)
Comment (felixfaisal) [reply pallets/ocex/src/lib.rs]: Done ✅

**Comment by serhii-temchenko** [comment]
Please fix clippy and all CI errors before moving PR to status `Ready for review`.

File: pallets/ocex/src/lib.rs [hunk 2]
```diff
@@ -768,45 +769,73 @@ pub mod pallet {
...
+									.is_ok()
+									{
+										processed_withdrawals
+											.try_push(withdrawal)
+											.unwrap_or_default();
+									}
+								}
+							}
+						}
+						// Not removing key from BtreeMap so that failed withdrawals can still be
```
Comment (felixfaisal) [inline pallets/ocex/src/lib.rs:891]: Push failed withdrawals back to the storage item

File: pallets/ocex/src/lib.rs [hunk 3]
```diff
@@ -768,45 +769,73 @@ pub mod pallet {
...
+						while withdrawal_vector.len() > 0 {
+							// Perform pop operation to ensure we do not leave any withdrawal left
+							// for a double spend
+							if let Some(withdrawal) = withdrawal_vector.pop() {
+								if let Some(converted_withdrawal) = withdrawal
+									.amount
+									.saturating_mul(Decimal::from(UNIT_BALANCE))
+									.to_u128()
+								{
+									if Self::transfer_asset(
```
Comment (felixfaisal) [inline pallets/ocex/src/lib.rs:870]: Emit failure event for the `else` branch

File: pallets/ocex/src/lib.rs (hunk 2 above)
Comment (felixfaisal) [reply pallets/ocex/src/lib.rs:891]: Updated ✅

File: pallets/ocex/src/lib.rs (hunk 3 above)
Comment (felixfaisal) [reply pallets/ocex/src/lib.rs:870]: Updated ✅

**Review by serhii-temchenko — DISMISSED** [review]

**Review by Gauthamastro — APPROVED** [review]

**Review by serhii-temchenko — APPROVED** [review]
Please merge only if CI is passed.

**Pull request merged**